### PR TITLE
Find most suitable city in the world

### PR DESCRIPTION
**Pull request by aya-seif**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -13,7 +13,12 @@ repositories {
 }
 
 dependencies {
+    implementation("junit:junit:4.13.1")
```
Comment (flutterdevtaha) [inline build.gradle.kts:16]: please remove these test implementations and make it like develop

File: src/main/kotlin/dataSource/utils/CsvParser.kt [hunk 2]
```diff
@@ -57,7 +57,7 @@ class CsvParser {
         return FoodPrices(
             loafOfFreshWhiteBread500g = tokenizedList.getFloat(ColumnIndex.LOAF_OF_FRESH_WHITE_BREAD_500G),
             riceWhite1kg = tokenizedList.getFloat(ColumnIndex.RICE_WHITE_1KG),
-            eggsRegular12 = tokenizedList.getFloat(ColumnIndex.EGGS_REGULAR_12),
+            eggsRegular12 = tokenizedList.getFloat(ColumnIndex.EGGS_REGULAR_12)!!,
```
Comment (flutterdevtaha) [inline src/main/kotlin/dataSource/utils/CsvParser.kt]: please revert changes here

File: src/main/kotlin/interactor/GetHighestSalaryAverageCititesNamesInteractor.kt [hunk 3]
```diff
@@ -9,14 +9,14 @@ class GetHighestSalaryAverageCititesNamesInteractor(
     fun execute(limit: Int): List<String> {
         return dataSource
             .getAllCitiesData()
-            .filter(::excludeNullSalariesAndLowQualityData)
+//            .filter(::excludeNullSalariesAndLowQualityData)
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetHighestSalaryAverageCititesNamesInteractor.kt:12]: revert it to orginal

File: src/main/kotlin/interactor/GetMostSuitableCityInteractor.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package interactor
+
+import model.CityEntity
+
+class GetMostSuitableCityInteractor {
+
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetMostSuitableCityInteractor.kt:6]: please complete the logic

File: src/main/kotlin/interactor/GetTopFashionCitiesNamesInteractor.kt [hunk 5]
```diff
@@ -12,21 +12,21 @@ class GetTopFashionCitiesNamesInteractor(
             .filter(::excludeNullPricesAndLowQualityData)
             .sortedBy { it.getAveragePrice() }
             .take(limit)
-            .map { it.cityName }
+            .map { it.cityName!! }
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetTopFashionCitiesNamesInteractor.kt:15]: revert changes

File: src/main/kotlin/model/CityEntity.kt [hunk 6]
```diff
@@ -8,7 +8,7 @@ data class CityEntity(
     val fruitAndVegetablesPrices: FruitAndVegetablesPrices,
     val foodPrices: FoodPrices,
     val servicesPrices: ServicesPrices,
-    val clothesPrices: ClothesPrices,
+    val clothesPrices: ClothesPrices?,
```
Comment (flutterdevtaha) [inline src/main/kotlin/model/CityEntity.kt:11]: revert changes

File: src/test/kotlin/GetMostSuitableCityTest.kt [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.function.Executable
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class GetMostSuitableCityTest {
+
+    private lateinit var getMostSuitableCityInteractor: GetMostSuitableCityInteractor
+    private lateinit var fakeDataSource: FakeDataSource
+
+    @BeforeEach
```
Comment (flutterdevtaha) [inline src/test/kotlin/GetMostSuitableCityTest.kt:17]: change it to only once to get an instance

**Review by flutterdevtaha — CHANGES_REQUESTED** [review]

File: src/main/kotlin/dataSource/utils/CsvParser.kt (hunk 2 above)
Comment (aya-seif) [reply src/main/kotlin/dataSource/utils/CsvParser.kt]: Done

File: src/main/kotlin/interactor/GetHighestSalaryAverageCititesNamesInteractor.kt (hunk 3 above)
Comment (aya-seif) [reply src/main/kotlin/interactor/GetHighestSalaryAverageCititesNamesInteractor.kt:12]: Done

File: src/main/kotlin/interactor/GetMostSuitableCityInteractor.kt (hunk 4 above)
Comment (aya-seif) [reply src/main/kotlin/interactor/GetMostSuitableCityInteractor.kt:6]: Done

File: src/main/kotlin/interactor/GetTopFashionCitiesNamesInteractor.kt (hunk 5 above)
Comment (aya-seif) [reply src/main/kotlin/interactor/GetTopFashionCitiesNamesInteractor.kt:15]: Done

File: src/main/kotlin/model/CityEntity.kt (hunk 6 above)
Comment (aya-seif) [reply src/main/kotlin/model/CityEntity.kt:11]: Done

File: src/test/kotlin/GetMostSuitableCityTest.kt (hunk 7 above)
Comment (aya-seif) [reply src/test/kotlin/GetMostSuitableCityTest.kt:17]: Done

**Pull request merged**